pkg/kubelet/status/status_manager.go

```diff
@@ -296,13 +296,14 @@ func (m *manager) GetPodResizeStatus(podUID types.UID) v1.PodResizeStatus {
 func (m *manager) SetPodAllocation(pod *v1.Pod) error {
 	m.podStatusesLock.RLock()
```

Comment:
Though this is not directly related to this issue, I wonder why a read lock is taken for writing a checkpoint.

Reply:
i believe, this is necessary because other methods write to this status, and this method reads from the status, so this would be necessary to avoid data race.